src/Microsoft.NetCore.Analyzers/UnitTests/Runtime/AvoidZeroLengthArrayAllocationsTests.cs

```diff
@@ -203,18 +236,48 @@ End Sub
-            VerifyBasicFix(
-                "Imports System\r\n" + arrayEmptySource + badSource,
-                "Imports System\r\n" + arrayEmptySource + fixedSource.Replace("System.Array.Empty", "Array.Empty"),
-                allowNewCompilerDiagnostics: true);
+
+            await new VerifyVB.Test
+            {
+                TestState =
+                {
+                    Sources = { arrayEmptySource + badSource },
+                    ExpectedDiagnostics =
+                    {
+                        GetBasicResultAt(7, 33, AvoidZeroLengthArrayAllocationsAnalyzer.UseArrayEmptyDescriptor, "Array.Empty(Of Integer)()"),
+                        GetBasicResultAt(8, 30, AvoidZeroLengthArrayAllocationsAnalyzer.UseArrayEmptyDescriptor, "Array.Empty(Of Byte)()"),
+                        GetBasicResultAt(9, 27, AvoidZeroLengthArrayAllocationsAnalyzer.UseArrayEmptyDescriptor, "Array.Empty(Of C)()"),
+                        GetBasicResultAt(13, 35, AvoidZeroLengthArrayAllocationsAnalyzer.UseArrayEmptyDescriptor, "Array.Empty(Of Integer())()"),
+                        GetBasicResultAt(14, 39, AvoidZeroLengthArrayAllocationsAnalyzer.UseArrayEmptyDescriptor, "Array.Empty(Of Integer()()())()"),
+                        GetBasicResultAt(16, 37, AvoidZeroLengthArrayAllocationsAnalyzer.UseArrayEmptyDescriptor, "Array.Empty(Of Integer(,))()")
+                    },
+                },
+                FixedState = { Sources = { arrayEmptySource + fixedSource } },
+                CodeFixValidationMode = CodeFixValidationMode.None,
+            }.RunAsync();
+
+            await new VerifyVB.Test
+            {
+                TestState =
+                {
+                    Sources = { "Imports System\r\n" + arrayEmptySource + badSource },
+                    ExpectedDiagnostics =
+                    {
+                        GetBasicResultAt(7 + 1, 33, AvoidZeroLengthArrayAllocationsAnalyzer.UseArrayEmptyDescriptor, "Array.Empty(Of Integer)()"),
+                        GetBasicResultAt(8 + 1, 30, AvoidZeroLengthArrayAllocationsAnalyzer.UseArrayEmptyDescriptor, "Array.Empty(Of Byte)()"),
+                        GetBasicResultAt(9 + 1, 27, AvoidZeroLengthArrayAllocationsAnalyzer.UseArrayEmptyDescriptor, "Array.Empty(Of C)()"),
+                        GetBasicResultAt(13 + 1, 35, AvoidZeroLengthArrayAllocationsAnalyzer.UseArrayEmptyDescriptor, "Array.Empty(Of Integer())()"),
+                        GetBasicResultAt(14 + 1, 39, AvoidZeroLengthArrayAllocationsAnalyzer.UseArrayEmptyDescriptor, "Array.Empty(Of Integer()()())()"),
+                        GetBasicResultAt(16 + 1, 37, AvoidZeroLengthArrayAllocationsAnalyzer.UseArrayEmptyDescriptor, "Array.Empty(Of Integer(,))()")
+                    },
+                },
+                FixedState = { Sources = { "Imports System\r\n" + arrayEmptySource + fixedSource.Replace("System.Array.Empty", "Array.Empty") } },
+                CodeFixValidationMode = CodeFixValidationMode.None,
```

Comment:
💡 Make sure we have a bug to review the code base for uses of `CodeFixValidationMode.None` once all tests are converted to the new test library.

Reply:
#2312